src/vscode-dts/vscode.d.ts

```diff
@@ -13992,7 +14010,29 @@ declare module 'vscode' {
 		 * @param uri Identifies the resource to open.
 		 * @returns A promise that resolves to a {@link TextDocument document}.
 		 */
-		export function openTextDocument(uri: Uri): Thenable<TextDocument>;
+		export function openTextDocument(uri: Uri, options?: {
+			/**
+			 * The {@link TextDocument.encoding encoding} of the document to use
+			 * for decoding the underlying buffer to text. If omitted, the encoding
+			 * will be guessed based on the file content and/or the editor settings
+			 * unless the document is already opened.
+			 *
+			 * Opening a text document that was already opened with a different encoding
+			 * has the potential of changing the text contents of the text document.
+			 * Specifically, when the encoding results in a different set of characters
+			 * than the previous encoding. As such, an error is thrown for dirty documents
+			 * when the specified encoding is different from the encoding of the document.
+			 *
+			 * See {@link TextDocument.encoding} for more information about valid
+			 * values for encoding. Using an invalid encoding will fallback to the
+			 * default encoding for the document.
+			 *
+			 * *Note* that if you open a document with an encoding that does not
+			 * support decoding the underlying bytes, content may be replaced with
+			 * substitution characters as appropriate.
```

Comment:
When can this happen and what does the substitution look like? Does the substitution cause any data loss?

Reply:
This can happen when you use an encoding that comes across byte sequences that have no mapping in that encoding, i.e. do not produce a valid character of the character set. In that case you would see characters like � or ? appearing in the editor. The actual visuals depend a bit on the editor but it can mean data loss because if you save the buffer back, you might change the amount of bytes per character and then the data is not recoverable, at least around that character (not the entire document).